### ISSUE TITLE
Language always overriden by diablo.ini even if locale is set to a different language

### ISSUE THREAD

**Issue by shantigilbert** [diasurgical/devilutionX#2677]
**Important information**
OS: EmuELEC / Version: ac0a6907685a71e4c9691d8f83bf9f3c5e6bc7fd / Compile it myself

**Describe**
Unless I am doing something very wrong, language always seems to be overridden by the one in diablo.ini.

**To Reproduce**
1. set language to es_MX (or whatever)

```
# locale
LANG=es_MX.UTF-8
LC_CTYPE="es_MX.UTF-8"
LC_NUMERIC="es_MX.UTF-8"
LC_TIME="es_MX.UTF-8"
LC_COLLATE="es_MX.UTF-8"
LC_MONETARY="es_MX.UTF-8"
LC_MESSAGES="es_MX.UTF-8"
LC_PAPER="es_MX.UTF-8"
LC_NAME="es_MX.UTF-8"
LC_ADDRESS="es_MX.UTF-8"
LC_TELEPHONE="es_MX.UTF-8"
LC_MEASUREMENT="es_MX.UTF-8"
LC_IDENTIFICATION="es_MX.UTF-8"
LC_ALL=es_MX.UTF-8

```
2. run 
```
# devilutionx --verbose
VERBOSE: Prefered locale: es_MX
VERBOSE: Best match locale: es
...
DEBUG: Couldn't open en.mo
DEBUG: Couldn't open en.gmo
en.gmo: No such file or directory
```
3. Diablo opens up in English

**Expected behavior**
DevilutionX should run with my selected locale and only fallback to English if no translation is available

**Additional context**
If I change diablo.ini to 
```
[Language]
Code=es
```
Diablo starts in Spanish as it should. 

EDIT: Also note that diablo.ini gets created automatically on the first run and the [LANGUAGE] block is based on the LOCALE at the time of creation but once it's created there is no way to change it unless you manually edit diablo.ini

**Comment by qndel** [comment]
> EDIT: Also note that diablo.ini gets created automatically on the first run and the [LANGUAGE] block is based on the LOCALE at
> the time of creation but once it's created there is no way to change it unless you manually edit diablo.ini

This sounds exactly like it should work for me lol, if the game didn't prioritize ini, then you wouldn't be able to have a language different than your system language

**Comment by shantigilbert** [comment]
So what if I change language after I run it the first time? Do I really have to manually edit the ini to change it? That is the whole point.

**Comment by qndel** [comment]
You can delete ini and let it regenerate itself? xD

**Comment by AJenbo** [comment]
> So what if I change language after I run it the first time?

How common is that scenario? From my experience it's much more common that people will have the OS run in there native local but prefer games in English. If they had to change there OS local to do that Windows users would have to pay MS for the Ultimate edition, and possibly reinstall Windows.

The behavior is trying to set a sane default, but allow the user full control.

Regarding editing ini with a controller, please see: https://github.com/diasurgical/devilutionX/issues/1561

**Comment by shantigilbert** [comment]
@AJenbo thanks for the reply, but like I said on my comment, I should have mentioned that this is an embedded system and not a PC. If by any chance you run Diablo before setting your language, you are stuck in English unless you manually change it, which again is not trivial like on a PC with keyboard. On the other hand I see that a menu is on the works which makes this issue irrelevant. Thank you for your time and work.

**Comment by AJenbo** [comment]
> I should have mentioned that this is an embedded system and not a PC. If by any chance you run Diablo before setting your language, you are stuck in English unless you manually change it, which again is not trivial

i think that way fairly clear, but: `not trivial < pay money to MS` especially when the user bases are skewed twords the later.

I have a similar device running RetroFW 2.0, here there is an editor that lets you edit ini files with out to much fuzz. But you can also edit the file from your pc and update it.

Is switching langauge in Diablo really something that you expect to be doing often when on the go?

**Comment by shantigilbert** [comment]
@AJenbo No, of course not. But the scenario I gave was if you ran it before setting your language, as many embedded systems are set to English by default, then the user is left with an unchangeable language, unless they manually edit the ini file, which I will repeat is not trivial in say, an odroid go advance, for an end user that has no idea that the ini takes priority in the first place. But I think I am overthinking it.

Anyways I have made a workaround for EmuELEC  and that is working for me. Thanks again for your time.
